pallets/rewards/src/lib.rs

```diff
@@ -0,0 +1,271 @@
+        Ok(())
+    }
+}
+
+decl_error! {
+    pub enum Error for Module<T: Config> {
+        /// Address was not found in the list of Itn addresses.
+        UnknownItnAddress,
+        /// Itn reward was already claimed.
+        ItnRewardAlreadyClaimed,
+        /// Provided signature was invalid.
+        InvalidSignature,
+        /// Balance can not be converted to a primitive.
+        UnableToCovertBalance,
+        // Insufficient balance to payout rewards.
+        InsufficientBalance,
+    }
+}
+
+decl_storage! {
+    trait Store for Module<T: Config> as Rewards {
+        /// Map of (Itn Address `T::AccountId`) -> (Reward `ItnRewardStatus`).
+        pub ItnRewards get(fn itn_rewards)
+            config(): map hasher(blake2_128_concat) T::AccountId
+            => Option<ItnRewardStatus<T::Balance>>;
+    }
+}
+
+decl_module! {
+    pub struct Module<T: Config> for enum Call where origin: <T as frame_system::Config>::Origin {
+        type Error = Error<T>;
+
+        fn deposit_event() = default;
+
+        /// Claim an ITN reward
+        ///
+        /// ## Arguments
+        /// `itn_address` - Address of the awarded account on ITN.
+        /// `signature` - Signature of the encoded reward data (originKey, ItnKey, amount)
+        ///
+        /// # Errors
+        /// * `InsufficientBalance` - Itn rewards has insufficient funds to issue the reward.
+        /// * `InvalidSignature` - `signature` had an invalid signer or invalid message.
+        /// * `ItnRewardAlreadyClaimed` - Reward issued to the `itn_address` has already been claimed.
+        /// * `UnknownItnAddress` - `itn_address` is not in the rewards table and has no reward to be claimed.
+        #[weight = (0, DispatchClass::Operational, Pays::No)]
+        pub fn claim_itn_reaward(origin, itn_address: T::AccountId, signature: T::OffChainSignature) -> DispatchResult {
+            let PermissionedCallOriginData{ sender, primary_did, .. } = <Identity<T>>::ensure_origin_call_permissions(origin.clone())?;
+            <ItnRewards<T>>::try_mutate(&itn_address, |reward| {
+                match reward{
+                    // Unclaimed. Attempt to claim.
+                    Some(ItnRewardStatus::UnClaimed(amount)) => {
+                        let amount = *amount;
+                        // `amount` and `bonded_amount` are equal in value but different types.
+                        // `deposit_amount` is 1 POLY more because we bond `bonded_amount`, we don't want all the users poly bonded.
+                        let (bonded_amount, deposit_amount) = Self::convert_balance(amount)?;
+                        ensure!(
+                            Self::balance() >= deposit_amount,
+                            Error::<T>::InsufficientBalance
+                        );
+                        ensure!(
+                            signature.verify((sender.clone(), itn_address.clone(), amount).encode().as_slice(), &itn_address),
```

Comment:
`signature.verify(reward_mainnet_address.clone(), itn_address)`
Should verify that signature = sign(reward_mainnet_address) via the itn_address